packages/client/src/features/change-bounds/move-element-key-listener.ts

```diff
@@ -0,0 +1,70 @@
+/********************************************************************************
+ * Copyright (c) 2025 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { KeyListener, matchesKeystroke, MoveElementRelativeAction, type Action, type GModelElement } from '@eclipse-glsp/sprotty';
+import type { SelectionService } from '../../base/selection-service';
+import { Grid } from '../grid/grid';
+import type { IChangeBoundsManager } from '../tools/change-bounds/change-bounds-manager';
+
+export class MoveElementKeyListener extends KeyListener {
+    constructor(
+        protected readonly selectionService: SelectionService,
+        protected readonly changeBoundsManager: IChangeBoundsManager,
+        protected readonly grid: Grid = Grid.DEFAULT
+    ) {
+        super();
+    }
+
+    override keyDown(_element: GModelElement, event: KeyboardEvent): Action[] {
+        const selectedElementIds = this.selectionService.getSelectedElementIDs();
```

Comment:
Need to be filtered so that only elements that have are actually moveable (`isMoveable` utility function) are moved.